lib/charty/colors/alpha_component.rb

```diff
@@ -0,0 +1,19 @@
+module Charty
+  module Colors
+    module AlphaComponent
+      attr_reader :a
+
+      def a=(a)
+        @a = if a.instance_of?(Integer)
+               check_range(a, 0..255, :a) / 255r
+             else
+               Rational(check_range(a, 0..1, :a))
+             end
```

Comment:
Can we use `canonicalize_component` here?

Reply:
Yes!